libs/components/modals/src/lib/modules/modal/modal-host.component.ts

```diff
@@ -109,15 +117,89 @@ export class SkyModalHostComponent implements OnDestroy {
       parent: this.injector,
     });
 
+    const modalsBefore = Array.from(
+      this.hostElement.nativeElement.parentElement.children
+    );
+
     const modalComponentRef = this.target.createComponent(
       factory,
       undefined,
       injector
     );
 
+    //const modalsAfter = this.hostElement.nativeElement.children;
+    const modalsAfter = Array.from(
+      this.hostElement.nativeElement.parentElement.children
+    );
+
+    // getting the element ref to the modal that was just opened
+    const modalElement = modalsAfter.filter(
+      (el) => modalsBefore.indexOf(el) < 0
+    )[0];
+
     modalInstance.componentInstance = modalComponentRef.instance;
 
+    // hidding all elements at the modal-host level from screenreaders
+    const ariaPreviousValueSiblings = new Map<Element, string | null>();
+    const siblings =
+      document.querySelector('sky-modal-host').parentElement.children;
+
+    for (let i = 0; i < siblings.length; i++) {
+      const sibling = siblings[i];
+
+      if (
+        sibling !== document.querySelector('sky-modal-host') &&
+        !sibling.hasAttribute('aria-live') &&
```

Comment:
i was basing this off of how angular handles `aria-hidden` with their modals. if we put aria-hidden on aria-live elements they won't announce and won't announce even after the modal is removed so it would be information that never gets to the user.